internal/servicefactory/interface.go

```diff
@@ -58,6 +61,10 @@ type ControllerServiceFactory interface {
 	// Access returns the access service. This includes the user and permission
 	// controller.
 	Access() *accessservice.Service
+	// SecretBackend returns the secret backend service.
+	SecretBackend(
+		clock.Clock, string, secretbackendservice.SecretProviderRegistry,
```

Comment:
Not sure we need to pass in a clock here. We can just initialise the service with the WallClock and have a helper in the service test package to pass in a test clock.